hack/build-go.sh

```diff
@@ -31,4 +31,4 @@ fi
 mkdir -p ${BIN_PATH}
 
 echo "Building ${REPO}/cmd/${WHAT} (${VERSION_OVERRIDE})"
-CGO_ENABLED=0 GOOS=${GOOS} GOARCH=${GOARCH} go build ${GOFLAGS} -ldflags "${GLDFLAGS}" -o ${BIN_PATH}/${WHAT} ${REPO}/cmd/${WHAT}
+CGO_ENABLED=1 GOOS=${GOOS} GOARCH=${GOARCH} go build ${GOFLAGS} -ldflags "${GLDFLAGS}" -o ${BIN_PATH}/${WHAT} ${REPO}/cmd/${WHAT}
```

Comment:
I made this conditional on the MCD only in https://github.com/openshift/machine-config-operator/pull/335/commits/c02eb5de6bcdff49931807070bbdd2cec2508087#diff-9283775b0feecc10455ec28bd08983b0.

Reply:
Hmm, actually that might not work. It looks like other components will then fail to build. From https://github.com/openshift/machine-config-operator/pull/335:

```
--> RUN WHAT=machine-config-controller ./hack/build-go.sh
Using version from git...
Building github.com/openshift/machine-config-operator/cmd/machine-config-controller (v3.11.0-589-gf62a5cf6-dirty)
vendor/github.com/coreos/go-systemd/sdjournal/functions.go...machine-config-operator/vendor/github.com/coreos/pkg/dlopen
error: build error: running 'WHAT=machine-config-controller ./hack/build-go.sh' failed with exit code 1
```

I'm not familiar enough with the golang build system, but it seems like it's trying to bundle all the vendored packages for all the targets instead of only the ones that actually need them?

Reply:
it's because the daemon package imports other packages (controller|server) and the build system bundle them together and that's why it's required for other components as well.

Reply:
You can use build tags to include / build files only when those tags are set...
https://golang.org/pkg/go/build/#hdr-Build_Constraints

Reply:
yeah, but that's gonna require some refactor as well

Reply:
my point is yeah, we're going to refactor for build tags but I guess for this PR we can ship with CGO_ENABLED=1 anyway, I'm creating a new issue to track this.

Reply:
https://github.com/openshift/machine-config-operator/issues/396